File: Day2_974_Subarray_ Sums_Divisible_by_K/Solution1.cs
public class Solution {
    public int SubarraysDivByK(int[] nums, int k) {
        int sum = 0;
        int count = 0;
        int[] hm = new int[k];
        hm[0] = 1;
        for(int i=0; i<nums.Length; i++){
            sum+=nums[i];
            int r = ((sum%k)+k)%k; // Same as var r = sum%k>= 0? sum%k : (sum%k)+k;  
            if(hm[r] > 0) count+=hm[r]; 
            hm[r]+=1;
        }
        return count;
    }
}

File: Day2_974_Subarray_ Sums_Divisible_by_K/Solution1.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var subarraysDivByK = function(nums, k) {
    var sum = 0;
    var hm = new Array(k);
    var count = 0;
    for(var i=0; i<nums.length; i++){
        sum+=nums[i];
        var r = sum%k;
        r = r>=0 ? r : r+k;  // same as int r = ((sum%k)+k)%k;      
        if(hm[r]) {
            count+=hm[r];
            hm[r]+=1;
        }
        else {
            hm[r] = 1;
        }
        if(r==0) count++;
    }
    return count
};
